src/main/java/in/projecteka/gateway/clients/ConsentRequestServiceClient.java

```diff
@@ -1,20 +1,64 @@
 package in.projecteka.gateway.clients;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import in.projecteka.gateway.common.Utils;
+import in.projecteka.gateway.common.cache.ServiceOptions;
 import in.projecteka.gateway.common.model.ErrorResult;
+import in.projecteka.gateway.registry.BridgeRegistry;
+import in.projecteka.gateway.registry.ServiceType;
+import in.projecteka.gateway.registry.YamlRegistryMapping;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
 import reactor.core.publisher.Mono;
 
+import java.time.Duration;
 import java.util.Map;
+import java.util.Optional;
+
+import static in.projecteka.gateway.common.Constants.TEMP_HIU_ID;
+
+@AllArgsConstructor
+public class ConsentRequestServiceClient implements ServiceClient {
+    private static final Logger logger = LoggerFactory.getLogger(ConsentRequestServiceClient.class);
+    private ServiceOptions serviceOptions;
+    private WebClient.Builder webClientBuilder;
+    private BridgeRegistry bridgeRegistry;
 
-public class ConsentRequestServiceClient implements ServiceClient{
     @Override
     public Mono<Void> routeRequest(Map<String, Object> request, String url) {
-        return null;
+        return Utils.serializeRequest(request)
+                .flatMap(serializedRequest ->
+                        webClientBuilder.build()
+                                .post()
+                                .uri(url + "/v1/consent-requests/init")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .bodyValue(serializedRequest)
+                                .retrieve()
+                                .onStatus(httpStatus -> !httpStatus.is2xxSuccessful(),
+                                        clientResponse -> Mono.error(ClientError.unableToConnect()))//TODO Errorhandling
+                                .bodyToMono(Void.class)
+                                .timeout(Duration.ofSeconds(serviceOptions.getTimeout())));
     }
 
     @Override
-    public Mono<Void> notifyError(ErrorResult request, String cmUrl) {
-        return null;
+    public Mono<Void> notifyError(ErrorResult request) {
+        //TODO check back when hiu id is dynamic
+        Optional<YamlRegistryMapping> config = bridgeRegistry.getConfigFor(TEMP_HIU_ID, ServiceType.HIU);
+        if (config.isEmpty()) {
+            logger.error("No mapping found for " + TEMP_HIU_ID);
```

Comment:
Why it's called temp in few places?

Reply:
Currently its fetched from the constants file. Angshuman was mentioning it can be dynamic in future. So called it Temp for now and there is a TODO for the same.